agenda/data/src/main/java/com/nibalk/tasky/agenda/data/remote/source/RetrofitRemoteEventDataSource.kt

```diff
@@ -72,15 +74,29 @@ class RetrofitRemoteEventDataSource(
     override suspend fun updateEvent(
         event: AgendaItem.Event
     ): Result<AgendaItem.Event?, DataError.Network> {
-        val formData = Json.encodeToString(event.toEventRequestDto())
+        Timber.d("[OfflineFirst-ImageIssue] REMOTE | Create EVENT (%s)", event.photos)
+        val formData = Json.encodeToString(event.toEventRequestForUpdateDto())
         val photoData = event.photos
-            .filterIsInstance<EventPhoto.Local>()
             .mapIndexedNotNull { index, eventPhoto ->
-                val bytes = Uri.parse(eventPhoto.localUri).getCompressedByteArray(context)
+                val photoBytes: ByteArray?
+                val photoKey: String
+                when(eventPhoto) {
+                    is EventPhoto.Local -> {
+                        photoKey = eventPhoto.key
+                        photoBytes = Uri.parse(eventPhoto.localUri).getCompressedByteArray(context)
+                    }
+                    is EventPhoto.Remote -> {
+                        photoKey = eventPhoto.key
+                        photoBytes = URL(eventPhoto.location).getCompressedByteArray()
+                    }
```

Comment:
There's no reason to consider the remote photos here, the server already knows them :D This could likely be the issue of your failed request, since if the server already has 4 photos stored and you now try to add another 4, but actually send them all again, so 8 in total, this will exceed the total photo count of 10 photos.

You can safely filter out the remote ones.